site/source/docs/porting/pthreads.rst

```diff
@@ -26,7 +26,10 @@ Special considerations
 
 The Emscripten implementation for the pthreads API should follow the POSIX standard closely, but some behavioral differences do exist:
 
-- If a page is built with the `-s USE_PTHREADS=1` linker flag, then it will not run backwards compatibly in a non-supporting browser. In order to enable backwards compatibility so that multithreaded pages can run in non-supporting browsers (except with pthread_create() disabled), pass the linker flag `-s USE_PTHREADS=2` instead. At runtime, you can use the `emscripten_has_threading_support()` function to test whether the current browser does have the capability to launch pthreads with `pthread_create()`. If a browser does not support threads, calls to `pthread_create()` will fail with error code `EAGAIN`.
+- At runtime, you can use the `emscripten_has_threading_support()` function to
+  test whether the current browser does have the capability to launch pthreads
+  with `pthread_create()`. If a browser does not support threads, calls to
+  `pthread_create()` will fail with error code `EAGAIN`.
```

Comment:
The function `emscripten_has_threading_support()` only functions in `USE_PTHREADS=2` mode, so if removing that, then in `USE_PTHREADS=1` builds `emscripten_has_threading_support()` will always return `true`, or the code will not reach a point where it can actually run. The above comment therefore is a bit misleading after `USE_PTHREADS=2` mode is removed. A better comment might be

```
- At runtime, you can use the `emscripten_has_threading_support()` function to test whether the currently executing code was compiled with pthreads support enabled. If this function returns true, then the currently executing code was compiled with multithreading support enabled (and the current browser supports multithreading).  If this function returns false, then the currently running code had been compiled with `-s USE_PTHREADS=0` linker flag.

If code is compiled with `-s USE_PTHREADS=1` and the current browser does not support multithreading via SharedArrayBuffer, then the page will not be able to load up, but throws an exception at loading time. In other words, it is not possible to build one binary 
that would be able to leverage multithreading when it is present, and gracefully fall back to singlethreaded execution when SharedArrayBuffer is not available. For such backwards compatibility, two separate builds must be done, one with -s USE_PTHREADS=1 and another one with -s USE_PTHREADS=0.
```